codice/ddf
# [2.26.x] DDF-6626 Adds initial implementation of trace context

#### What does this PR do?
Adds a trace context to incoming requests.  Updates the SecurityLogger to include the trace-id to allow for audit event correlation.

#### Who is reviewing it? 
@pklinef @derekwilhelm @glenhein 

#### Select relevant component teams: 
@codice/io 
@codice/security 


#### Ask 2 committers to review/merge the PR and tag them here.
@pklinef @derekwilhelm @glenhein 


#### How should this be tested?
1) Build/install
2) Install ddf-ui
3) Log into Intrigue and perform some actions like executing a search, downloading a product, etc.
3) Verify in the security log that a trace-id is included in all audit events from the user initiated actions.  Note that any actions executed by the server and not as a direct result of a client request will not include a trace-id.

#### Any background context you want to provide?
This is an initial implementation in support of the W3C trace context specification https://www.w3.org/TR/trace-context/

#### What are the relevant tickets?
Fixes: #6626 

#### Screenshots


#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
